### PR TITLE
Movie service and controller created.

### PR DESCRIPTION
**Pull request by Kuleb9lka**

### CODE REVIEW

File: src/main/java/by/library/controller/MovieController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @Autowired
+    private MovieService movieService;
+
+    @GetMapping
+    public List<Movie> getAll(){
+
+        return movieService.getAll();
+    }
+
+    @GetMapping("/get/{id}")
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/controller/MovieController.java:28]: remowe path parameter get only {id}

File: src/main/java/by/library/controller/MovieController.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    public Movie get(@PathVariable Long id){
+
+        return movieService.get(id);
+    }
+
+    @GetMapping("/create/{name}/{date}")
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/controller/MovieController.java:34]: get is not for creation

File: src/main/java/by/library/controller/MovieController.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    public Movie create(@PathVariable String name, @PathVariable Timestamp date) {
+
+        Movie movie = new Movie(name, date);
+
+        return movieService.create(movie);
+    }
+
+    @GetMapping("/update/{id}/{name}/{date}")
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/controller/MovieController.java:42]: Get is not for updating

File: src/main/java/by/library/controller/MovieController.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        Movie movie = movieService.get(id);
+
+        movie.setName(name);
+
+        movie.setDate(date);
+
+        return movieService.update(movie);
+    }
+
+    @GetMapping("/delete/{id}")
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/controller/MovieController.java:54]: get is not for deleting

File: src/main/java/by/library/model/movie/Movie.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import lombok.*;
+
+import javax.persistence.*;
+import java.io.Serializable;
+import java.sql.Timestamp;
+
+@Data
+@NoArgsConstructor
+@RequiredArgsConstructor
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/model/movie/Movie.java:12]: do you realy need this?

File: src/main/java/by/library/model/movie/Movie.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Long id;
+
+    @Column(name = "name")
+    @NonNull private String name;
+
+    @Column(name = "date_time")
+    @NonNull private Timestamp date;
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/model/movie/Movie.java:27]: use LocalDateTime

File: src/main/java/by/library/model/ticket/Ticket.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Long id;
+
+    @Column(name = "user_id")
+    private int userId;
+
+    @Column(name = "movie_id")
+    private int movieId;
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/model/ticket/Ticket.java:25]: use Movie object

File: src/main/java/by/library/service/impl/MovieServiceImpl.java [hunk 8]
```diff
@@ -35,26 +35,28 @@ public Movie get(Long id) {
     }
 
     @Override
-    public Movie create(Movie movie) {
+    public Movie create(MovieDto movie) {
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/service/impl/MovieServiceImpl.java:38]: may be right naming for dto is movieDto?

File: src/main/java/by/library/service/impl/MovieServiceImpl.java [hunk 9]
```diff
@@ -35,26 +35,28 @@ public Movie get(Long id) {
...
 
-        movieRepository.save(movie);
+        Movie movie1 = new Movie(movie.getName(), movie.getDateTime());
 
-        return movie;
+        movieRepository.save(movie1);
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/service/impl/MovieServiceImpl.java:42]: if you give right name for dto you can rename this like movie

Comment (Kuleb9lka) [reply src/main/java/by/library/service/impl/MovieServiceImpl.java]: +

File: src/main/java/by/library/controller/UserController.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+//    @GetMapping("/get-user")
+//    public User getUser(){
+//        return new User(1, "vasya", "dsfsdf", "sdfsdfsdf", "male", 12);
+//    }
+//
+//    @PostMapping("/save")
+//    public void addUser(@RequestBody User user){
+//
+//        System.out.println(user);
+//    }
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/controller/UserController.java:45]: remove this one if not using

File: src/main/java/by/library/service/impl/MovieServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+
+    private final MovieRepository movieRepository;
+
+    @Autowired
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/service/impl/MovieServiceImpl.java:18]: it is optinal annoation in this case

File: src/main/java/by/library/service/impl/MovieServiceImpl.java (hunk 9 above)
Comment (Maksim-maevsky) [reply src/main/java/by/library/service/impl/MovieServiceImpl.java]: do you stay movie1 insted rename to movie

File: src/main/java/by/library/controller/UserController.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+        Optional<User> optionalUser = userRepository.findByLogin("bogdan");
+
+        if (optionalUser.isPresent()) {
+
+            User user = optionalUser.get();
+        }
+
+        return userRepository.findAll();
+    }
+
```
Comment (Maksim-maevsky) [inline src/main/java/by/library/controller/UserController.java:34]: why you dont use service, it is not good idea use repository directly from controller

File: src/main/java/by/library/controller/UserController.java (hunk 10 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/controller/UserController.java:45]: done

File: src/main/java/by/library/service/impl/MovieServiceImpl.java (hunk 11 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/service/impl/MovieServiceImpl.java:18]: done

File: src/main/java/by/library/controller/MovieController.java (hunk 1 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/controller/MovieController.java]: done

File: src/main/java/by/library/controller/MovieController.java (hunk 2 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/controller/MovieController.java]: done

File: src/main/java/by/library/controller/MovieController.java (hunk 3 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/controller/MovieController.java]: done

File: src/main/java/by/library/controller/MovieController.java (hunk 4 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/controller/MovieController.java]: done

File: src/main/java/by/library/model/movie/Movie.java (hunk 5 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/model/movie/Movie.java]: done

File: src/main/java/by/library/model/movie/Movie.java (hunk 6 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/model/movie/Movie.java]: done

File: src/main/java/by/library/model/ticket/Ticket.java (hunk 7 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/model/ticket/Ticket.java]: done

File: src/main/java/by/library/service/impl/MovieServiceImpl.java (hunk 8 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/service/impl/MovieServiceImpl.java]: done

File: src/main/java/by/library/service/impl/MovieServiceImpl.java (hunk 9 above)
Comment (Kuleb9lka) [reply src/main/java/by/library/service/impl/MovieServiceImpl.java]: done

**Pull request merged**